client_code/MultiSelectDropDown/__init__.py

```diff
@@ -208,6 +216,29 @@ def visible(self, val):
         )
         _HtmlPanel.visible.__set__(self, val)
 
+    @property
+    def width(self):
+        return self._width
+
+    @width.setter
+    def width(self, value):
+        if value == "":
+            value = None
+
+        _inner_dom_node = self._dom_node.children()
+        if value is not None:
+            if value.replace(".", "").isnumeric():
+                self._dom_node.css("width", f"{value}px")
+                _inner_dom_node.css("width", f"{value}px")
+            else:
+                self._dom_node.css("width", value)
+                _inner_dom_node.css("width", value)
+        else:
+            _inner_dom_node.css("width", "fit-content")
+            self._dom_node.css("width", "fit-content")
+
+        self._width = value
+
```

Comment:
```suggestion
    width = _component_property("width", "data-width", lambda v: v or "auto")
```

note that this property won't be dynamic - but that's a separate issue for a separate pr

Reply:
This would be simpler, but if we can make it dynamic (which my current implementation does) so that when the width property is used it can be changed, why not? I very well might be missing another reason that we might prefer the simpler static option to the more verbose dynamic one, but I'm not sure what that is right now.

Reply:
To make it dynamic is a one line change on my part. 

Certain properties like `enable_select_all` would also benefit from this change. It's a bug that I've not gotten around to fixing because no-one's complained. 

I'm happy to make that PR and for you to rebase on top of it. 

The Multiselect library we're using already has an api for defining the width so we should use it.

The short version is abstracted away and uses the library's api to implement options that the library already has an api for. I can't see a justification for not using it.

Reply:
If its not too much trouble to you, I would like to make this dynamic -- seems like this is a property that's more prone to change than `enable_select_all` imho

Reply:
When you say dynamic are we talking about the same thing?

I mean you can set it in code after the component has been instantiated.

Reply:
When I say dynamic, I mean that you can change the value through some piece of code after the component is first instantiated. With my OG implementation you can do that, but with the native component's property, I can't change the width after setting it initially.